src/components/TaxPicker.tsx

```diff
@@ -23,60 +23,44 @@ type TaxPickerProps = {
     insets?: EdgeInsets;
 
     /** Callback to fire when a tax is pressed */
-    onSubmit: () => void;
+    onSubmit: (tax: ListItem) => void;
+
+    /** Default key of selectedtax */
+    defaultTaxKey?: string;
 };
 
-function TaxPicker({selectedTaxRate = '', taxRates, insets, onSubmit}: TaxPickerProps) {
-    const styles = useThemeStyles();
+function TaxPicker({selectedTaxRate = '', taxRates, insets, onSubmit, defaultTaxKey}: TaxPickerProps) {
     const StyleUtils = useStyleUtils();
     const {translate} = useLocalize();
     const [searchValue, setSearchValue] = useState('');
 
-    const taxRatesCount = TransactionUtils.getEnabledTaxRateCount(taxRates.taxes);
-    const isTaxRatesCountBelowThreshold = taxRatesCount < CONST.TAX_RATES_LIST_THRESHOLD;
+    if (!taxRates) {
+        return;
+    }
 
-    const shouldShowTextInput = !isTaxRatesCountBelowThreshold;
+    const taxRatesCount = TransactionUtils.getEnabledTaxRateCount(taxRates?.taxes);
+    const isTaxRatesCountBelowThreshold = taxRatesCount < CONST.TAX_RATES_LIST_THRESHOLD;
 
-    const selectedOptions = useMemo(() => {
-        if (!selectedTaxRate) {
-            return [];
-        }
+    const defaultSelectedTaxKey = defaultTaxKey ?? taxRates.defaultExternalID;
 
-        return [
-            {
-                name: selectedTaxRate,
-                enabled: true,
-                accountID: null,
-            },
-        ];
-    }, [selectedTaxRate]);
+    const getModifiedName = (data: TaxRate, code: string) => `${data.name} (${data.value})${defaultSelectedTaxKey === code ? ` • ${translate('common.default')}` : ''}`;
 
-    const sections = useMemo(() => {
-        const {taxRatesOptions} = OptionsListUtils.getFilteredOptions({}, {}, [], searchValue, selectedOptions, [], false, false, false, {}, [], false, {}, [], false, false, true, taxRates);
-        return taxRatesOptions;
-    }, [taxRates, searchValue, selectedOptions]);
+    const shouldShowTextInput = !isTaxRatesCountBelowThreshold;
 
-    const selectedOptionKey = sections?.[0]?.data?.find((taxRate) => taxRate.searchText === selectedTaxRate)?.keyForList;
+    const section = Object.entries(taxRates.taxes ?? {}).map(([key, item]) => ({text: getModifiedName(item, key), keyForList: key}));
+    const selectedOptionKey = section.find((taxRate) => taxRate.text === selectedTaxRate)?.keyForList;
```

Comment:
NAB: I don't like comparing it by text, not by keys. But I can see this was introduced earlier by someone else, we are just switching to `SelectionList`.

Reply:
Yes, I am currently working on this, just need to update one IOU page where we use TaxPicker too

Reply:
yep I agree that its not good so if we can change this to compare the codes it would be better

Reply:
Done!